### PR TITLE
EdgeAgent: Add support to get Module logs

### PR DESCRIPTION
**Pull request by varunpuranik**
This PR adds support to get logs for modules in the EdgeAgent. 
The next step will be hook up this to streaming/uploading the logs.

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/RuntimeInfoProvider.cs [hunk 1]
```diff
@@ -60,10 +61,22 @@ public async Task<IEnumerable<ModuleRuntimeInfo>> GetModules(CancellationToken c
             Option<ContainerInspectResponse> edgeAgentReponse = await this.GetEdgeAgentContainerAsync();
             edgeAgentReponse.ForEach(e => containerInspectResponses.Add(e));
 
-            List<ModuleRuntimeInfo> modules = containerInspectResponses.Select(c => InspectResponseToModule(c)).ToList();
+            List<ModuleRuntimeInfo> modules = containerInspectResponses.Select(InspectResponseToModule).ToList();
             return modules;
         }
 
+        public Task<Stream> GetModuleLogs(string module, bool follow, Option<int> tail, CancellationToken cancellationToken)
```
Comment (myagley) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/RuntimeInfoProvider.cs]: We may want to add a `LoggingOptions` struct for the `follow` and `tail` fields. I don't think it's necessarily a problem now, but this type of thing does have a tendency to introduce parameter bloat.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/ModuleManagementHttpClient.cs [hunk 2]
```diff
@@ -20,7 +21,7 @@ public ModuleManagementHttpClient(Uri managementUri, string serverSupportedApiVe
             Preconditions.CheckNotNull(managementUri, nameof(managementUri));
             Preconditions.CheckNonWhiteSpace(serverSupportedApiVersion, nameof(serverSupportedApiVersion));
             Preconditions.CheckNonWhiteSpace(clientSupportedApiVersion, nameof(clientSupportedApiVersion));
-            this.inner = this.GetVersionedModuleManagement(managementUri, serverSupportedApiVersion, clientSupportedApiVersion);
+            this.inner = GetVersionedModuleManagement(managementUri, serverSupportedApiVersion, clientSupportedApiVersion);
```
Comment (myagley) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/ModuleManagementHttpClient.cs]: Is this no longer a method? (missing `this`)

Comment (varunpuranik) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/ModuleManagementHttpClient.cs]: I made it static

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/RuntimeInfoProvider.cs (hunk 1 above)
Comment (varunpuranik) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/RuntimeInfoProvider.cs]: Yes, I agree. I am adding more such types in the actual logs processing. Will use it here if needed as well.

**Pull request merged**